lbz/events/api.py

```diff
@@ -65,13 +82,20 @@ def get_all_failed_events(self) -> List[BaseEvent]:
         return self._failed_events
 
     def send(self) -> None:
+        self._sent_events = []
+        self._failed_events = []
+
         try:
-            if self._pending_events:
-                entries = [self._create_eb_entry(event) for event in self._pending_events]
+            while self._pending_events:
+                events = self._pending_events[:MAX_EVENTS_TO_SEND_AT_ONCE]
+                entries = [self._create_eb_entry(event) for event in events]
                 client.eventbridge.put_events(Entries=entries)
-            self._mark_sent()
+
+                self._sent_events.extend(events)
+                self._pending_events = self._pending_events[MAX_EVENTS_TO_SEND_AT_ONCE:]
         except Exception as err:
-            self._mark_failed()
+            self._failed_events.extend(self._pending_events)
+            self._pending_events = []
```

Comment:
Why would You empty that list if the events were not send and they possible could be?
That would make debugging harder and b. that would not allow tu build retry logic around that method. the line 
`self._pending_events = self._pending_events[MAX_EVENTS_TO_SEND_AT_ONCE:]`
should be used here and then I would add a flag either on Init level or on method level `raise_at_first_sending_failure` +/- and only if that flag is present do actual raise on first failure in other case send remaining and not raise we could also try doing it a 3 way switch with
a) raise at first (and the the current logic makes more sense)
b) raise after all if there was an error 
c) never raise if anything was send (raise only when nothing was send and something should be)

Reply:
So, this implementation is the **A)** scenario, and I empty this list here only because all of its events are marked as failed above. I did it this way to stay compatible with our previous agreements that the `send()` method does not leave anything pending 😉 

Also, our previous agreements touched the retrying logic - we decided it should be implemented outside this library if anyone needs it, and with this code, still, it is possible, based on the `event_api.failed_events` list. 

Debugging shouldn't be a problem as well, as still, the error is raised, nothing will be hidden 😉

Reply:
I was considering the **B)** scenario for a moment but I thought that getting multiple errors from a single function may produce more unnecessary chaos. But we can talk about it and go this way 👍 

But I would be against giving the possibility to choose because we talk about the situations that should never happen, and if happen, they need to be fixed as soon as possible 😉

Reply:
As we agreed outside this PR, I have changed the strategy to always try to send all registered events 🙂